src/components/new-suggestion/suggestion-page-section.tsx

```diff
@@ -0,0 +1,155 @@
+import ReviewArea from "@/components/new-suggestion/areas/review.area"
+import { Area } from "@/constants/area"
+import { FormProvider, useForm } from "react-hook-form"
+import {
+  InputsSongInformation,
+  NewSuggestion,
+  NewSuggestionInstrument,
+} from "@/interfaces/new-suggestion"
+import InstrumentsArea from "@/components/new-suggestion/areas/instruments/instruments.area"
+import { Database } from "@/types/database"
+import { Instrument } from "@/types/database-types"
+import { useSupabaseClient } from "@supabase/auth-helpers-react"
+import { getInstruments } from "@/services/instrument.service"
+import Spinner from "@/components/utils/spinner"
+import ErrorPopup from "@/components/popups/error-popup"
+import { submitSongInformationForm } from "@/helpers/new-suggestion.helper"
+
+interface SuggestionPageSectionProps {
+  title: string
+  newSuggestion: NewSuggestion
+  startingArea: Area
+  onCloseClicked(): void
+  onSongInformationSubmit?(songInformation: InputsSongInformation): void
+  onInstrumentSubmit?(newInstruments: NewSuggestionInstrument[]): void
+  onAreaSelect?(newArea: Area): void
+  onReviewSubmit(onSuccess: () => void, onError: () => void): void
+}
+
+const SuggestionPageSection = ({
+  title,
+  newSuggestion,
+  startingArea,
+  onReviewSubmit,
+  onAreaSelect,
+  onCloseClicked,
+  onInstrumentSubmit,
+  onSongInformationSubmit,
+}: SuggestionPageSectionProps) => {
+  const [activeArea, setActiveArea] = useState<Area>(startingArea)
+  const supabaseClient = useSupabaseClient<Database>()
+  const [showSpinner, setShowSpinner] = useState<boolean>(false)
+  const [showLoadingError, setShowLoadingError] = useState<boolean>(false)
+
+  const methods = useForm<InputsSongInformation>({
+    defaultValues: {
+      ...newSuggestion,
+      artist: newSuggestion.artist.join(","),
+    } as InputsSongInformation,
+    shouldFocusError: false,
+  })
+
+  const [instrumentList, setInstrumentList] = useState<Instrument[]>([])
+  const [newSuggestionInstruments, setNewSuggestionInstruments] = useState<
+    NewSuggestionInstrument[]
+  >(newSuggestion.instruments)
+
+  useEffect(() => {
+    setShowSpinner(true)
+    getInstruments(supabaseClient)
+      .then((response) => {
+        if (response.error || response.data?.length === 0) {
+          setShowLoadingError(true)
+          return
+        }
+
+        setInstrumentList(response.data as Instrument[])
+      })
+      .catch(() => {
+        setShowLoadingError(true)
+      })
+      .finally(() => {
+        setShowSpinner(false)
+      })
+  }, [supabaseClient])
+
+  const handleAreaChange = (area: Area) => {
+    if (onAreaSelect) onAreaSelect(area)
+
+    submitSongInformationForm()
+    if (onSongInformationSubmit) {
+      onSongInformationSubmit(methods.getValues())
+    }
+
+    if (area !== Area.Instruments)
+      if (onInstrumentSubmit) onInstrumentSubmit(newSuggestionInstruments)
+
+    setActiveArea(area)
```

Comment:
I'm very confused about what is happening here. Especially the if statements got me dazzled. Could you explain?

Reply:
onInstrumentSubmit and onSongInformation are optional objects. Will change them to required to remove optionality.